src/web/common.rkt

```diff
@@ -37,49 +48,55 @@
   `(div ,name ": " (span ([class "number"]
                           ,@(if title `([title ,title]) '()))
                          ,@values)))
-
-;; TODO(interface): currently program->c doesn't take the repr into account
+  
 (define languages
-  `(("TeX" . ,core->tex) 
+  `(("TeX" . ,core->tex)
+    ("FPCore" . ,fpcore->string)
     ("C" . ,(curryr core->c "code"))))
 
 (define (render-program #:to [result #f] test)
   (define output-prec (test-output-prec test))
-  (define output-repr (get-representation output-prec))
-
   (define in-prog (program->fpcore (resugar-program (test-program test) output-prec)))
   (define out-prog (and result (program->fpcore (resugar-program result output-prec))))
 
+  (define in-prog* (fpcore-add-props in-prog (list ':precision output-prec)))
+  (define out-prog* (and out-prog (fpcore-add-props out-prog (list ':precision output-prec))))
+
   (define versions
     (reap [sow]
       (for ([(lang converter) (in-dict languages)])
-        (when (and (fpcore? in-prog) (or (not out-prog) (fpcore? out-prog)))
-          (sow (cons lang (cons (converter in-prog)
-                                (and out-prog (converter out-prog)))))))))
+        (let ([ext (string-downcase lang)]) ; FPBench organizes compilers by extension
+          (when (and (fpcore? in-prog*) (or (not out-prog*) (fpcore? out-prog*))
+                    (or (equal? ext "fpcore")                           
+                        (and (supported-by-lang? in-prog* ext) ; must be valid in a given language  
+                             (or (not out-prog*) (supported-by-lang? out-prog* ext)))))
+            (sow (cons lang (cons (converter in-prog*)
+                                  (and out-prog* (converter out-prog*)))))
+    )))))
 
   `(section ([id "program"])
      ,(if (equal? (program-body (test-precondition test)) 'TRUE)
           ""
           `(div ([id "precondition"])
              (div ([class "program math"])
-                  "\\[" ,(expr->tex (program-body (test-precondition test))) "\\]")))
+                  "\\[" ,(expr->tex (resugar-program (program-body (test-precondition test)) output-prec)) "\\]")))
```

Comment:
Ugh, does `resugar-program` still use precisions instead of representations? Leave this as is, but we should fix that later.